cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/DatastoreManager.java

```diff
@@ -333,7 +333,7 @@ private Datastore createDatastore(String dbName, KeyProvider provider) throws Da
             // if it does not exist
 
             //Pass database directory, database name, and SQLCipher key provider
-            DatastoreImpl ds = new DatastoreImpl(dbDirectory, dbName, provider);
+            QueryableDatastore ds = new QueryableDatastore(dbDirectory, dbName, provider);
```

Comment:
As you know I am in favour of the composition of the implementations. I think when we were in discussion the `getQueryableDatastore().find()` method was seen as adding too much complexity for users (although it is still simpler than the current `IndexManager`.

The facade pattern seems like a reasonable compromise, keeping the interface simple for the user as `Datastore` with all available methods, but the complexity of implementation broken down into different internal types.
I guess this would look like `DatabaseImpl` constructing the various instances required (e.g. `QueryableDatastore` and delegating the interface calls. If we wanted to reduce the boiler plate we could use the `java.lang.reflect.Proxy` to direct to the necessary objects, but obviously that would introduce complexity of its own, for example, in identifying which method needed to be invoked where.